### PR TITLE
Fix: Added error handling when fetch users api call fails

### LINKED ISSUE #2103: Non-Error promise rejection captured with keys: _meta, data, error, status
Sentry Issue: [MANAGER-UI-G5](https://zestyio.sentry.io/issues/2277454417/?referrer=github_integration)

```
UnhandledRejection: Non-Error promise rejection captured with keys: _meta, data, error, status
```

### PR DESCRIPTION
**Pull request by theofficialnar**
Added a catch when api call fails

Closes #2103 

image

### CODE REVIEW

**Comment by theofficialnar** [comment]
> @theofficialnar How does this fix the reported error?
> 
> The error that we are seeing is indicating that a promise was rejected, but instead of an Error object, which is the standard practice, an object with the keys _meta, data, error, and status was provided.
> 
> Which tells me its not really a unhandled rejection error but a handled one that is reading something unexpected

@agalin920 right, I realized the promise was throwing the `res` object instead of an Error. I'll update that. But doesn't it also make sense to add that catch at the end to handle that thrown error?

**Comment by theofficialnar** [comment]
> @theofficialnar yea I agree. Just pointing out a potentially deeper issue

@agalin920 I found out that `request.js` doesn't throw an error for 400 responses. So, I went ahead and made it throw a new error when res.status is 4**. This makes it so that the promise is properly rejected when res.status is 4** and the redux actions utilizing `request` will then be able to properly handle the failed api call via catch. 

It's probably worth mentioning that I am not 100% sure why `request.js` was made to not throw an error for 4** responses to begin with, since I can see that there are some commented out code that were supposedly handling those.

File: src/utility/request.js [hunk 1]
```diff
@@ -75,6 +75,11 @@ export function request(url, opts = {}) {
 
       // if (res.status === 422) {}
 
+      // Client error
+      if (res.status >= 400 && res.status < 500) {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/utility/request.js:79]: Gotcha, I guess that answers my question as to why `request.js` doesn't throw errors for 400s then. I'll just handle that inside `fetchUsers()` then.

**Comment by theofficialnar** [comment]
@agalin920 I see, right I'll add some handling for that as well. There is an event under this similar sentry issue that doesn't have a 403 on /instances though, and only 403d on /domains and /users. Since domains isn't throwing the response on 403 but /users does, don't you think it's also worth handling that (which is what I currently have added on the changes)?

**Review by shrunyan — APPROVED** [review]

**Pull request merged**